### PR TITLE
Enable cargo-spellcheck for blockchain/state_manager 

### LINKED ISSUE #1690: Enable `cargo-spellcheck` for `blockchain/state_manager`
**Issue summary**

Make sure that `cargo spellcheck` can run without errors in `blockchain/state_manager`. For details, see the tracking issue.

**Other information and links**

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes #1690

**Other information and links**

### CODE REVIEW

File: blockchain/state_manager/src/lib.rs [hunk 1]
```diff
@@ -153,28 +153,28 @@ where
         &self.chain_config
     }
 
-    /// Gets actor from given [Cid], if it exists.
+    /// Gets actor from given `[Cid]`, if it exists.
```
Comment (lemmih) [inline blockchain/state_manager/src/lib.rs:156]: These three things are different:

    [Cid]         -- Creates a link to Cid
    `[Cid]`       -- Code snippet containing "[Cid]"
    [`Cid`]       -- Code snippet that links to Cid

You want the last one. When in doubt, run `cargo doc --open` and make sure the documentation looks right.

Comment (lemmih) [reply blockchain/state_manager/src/lib.rs:156]: This should be fixed everywhere.

File: blockchain/state_manager/src/lib.rs [hunk 2]
```diff
@@ -1291,7 +1291,7 @@ where
         Ok(out)
     }
 
-    /// Similar to `resolve_to_key_addr` in the vm crate but does not allow `Actor` type of addresses.
+    /// Similar to `resolve_to_key_addr` in the VM crate but does not allow `Actor` type of addresses.
```
Comment (lemmih) [inline blockchain/state_manager/src/lib.rs:1294]: This should be `forest_vm` crate. It's not short for virtual machine, it's an actual name.

**Review by lemmih — CHANGES_REQUESTED** [review]
Good start. Caught 3 typos.

The extra dictionary should be as small as possible. We'll eventually need to define each word in the dictionary as part of our documentation.

File: .config/forest.dic [hunk 3]
```diff
@@ -11,3 +11,9 @@ APIs
 CBOR
 codec
 BLAKE2b
+VM
+multisig
+tuple
+mutex
+RPC
+parameterized
```
Comment (lemmih) [inline .config/forest.dic:19]: This word should already be in the built-in dictionary. Is it missing for you?

Comment (tyshko5) [reply .config/forest.dic:19]: It is

Comment (lemmih) [reply .config/forest.dic:19]: Hmm, that's a big problem. I'll look into how the dictionaries can be different.

File: blockchain/chain/src/store/chain_store.rs [hunk 4]
```diff
@@ -842,7 +842,7 @@ where
     })
 }
 
-/// Returns messages from key-value store based on a slice of [Cid]s.
+/// Returns messages from key-value store based on a slice of ['Cid']s.
```
Comment (lemmih) [inline blockchain/chain/src/store/chain_store.rs:845]: Run `cargo doc --open` and check if this is right.

**Review by lemmih — CHANGES_REQUESTED** [review]

File: .config/forest.dic (hunk 3 above)
Comment (lemmih) [reply .config/forest.dic]: Fixed.

**Review by lemmih — APPROVED** [review]

File: blockchain/state_manager/src/chain_rand.rs [hunk 5]
```diff
@@ -112,7 +112,7 @@ where
             .await
     }
 
-    /// network v0-12; with lookback
+    /// network version 0-12; with look-back
```
Comment (LesnyRumcajs) [inline blockchain/state_manager/src/chain_rand.rs:115]: are you sure about this `look-back`? In the Filecoin spec it's a one word. https://spec.filecoin.io/

Comment (lemmih) [reply blockchain/state_manager/src/chain_rand.rs:115]: They're quite fast and loose about creating new words. Lookback isn't a word you would find in a dictionary but rather the combination of two separate words. Ideally I would want the dictionary to only contain new words that have a specific meaning in the filecoin community (like 'Filecoin', 'Merkel', 'tipset', etc).

Comment (LesnyRumcajs) [reply blockchain/state_manager/src/chain_rand.rs:115]: Fine with me.

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**